lib/spack/spack/database.py

```diff
@@ -309,7 +309,11 @@ def reindex(self, directory_layout):
                 for spec in directory_layout.all_specs():
                     # Create a spec for each known package and add it.
                     path = directory_layout.path_for_spec(spec)
-                    self._add(spec, path, directory_layout)
+                    old_info = old_data.get(spec.dag_hash())
+                    explicit = False
```

Comment:
I wonder if explicit should be `True` by default. Right now this information is not used at all, but if we implement something like : 
```
spack uninstall clean  
```
having the default to `True` will be more conservative. I don't have a strong opinion on this though, just want to share a doubt...